File: Dumplings/src/main/java/net/sinclairstudios/dumplings/ui/widgets/DismissArrayAdapterItemListViewTouchListener.java
package net.sinclairstudios.dumplings.ui.widgets;

import android.widget.ArrayAdapter;
import android.widget.ListView;
import com.example.android.swipedismiss.SwipeDismissListViewTouchListener;

public class DismissArrayAdapterItemListViewTouchListener extends SwipeDismissListViewTouchListener {

    public DismissArrayAdapterItemListViewTouchListener(ListView listView, ArrayAdapter<?> arrayAdapter) {
        super(listView, dismissCallbacksForArrayAdapter(arrayAdapter));
    }

    private static <T> DismissCallbacks dismissCallbacksForArrayAdapter(final ArrayAdapter<T> arrayAdapter) {
        return new DismissCallbacks() {
            @Override
            public boolean canDismiss(int position) {
                return position < arrayAdapter.getCount();
            }

            @Override
            public void onDismiss(ListView listView, int[] reverseSortedPositions) {
                for (int position : reverseSortedPositions) {
                    arrayAdapter.remove(arrayAdapter.getItem(position));
                }
                arrayAdapter.notifyDataSetChanged();
            }
        };
    }
}

File: app/src/main/java/net/sinclairstudios/dumplings/widget/DismissArrayAdapterItemListViewTouchListener.kt
package net.sinclairstudios.dumplings.widget

import android.widget.ArrayAdapter
import android.widget.ListView
import swipedismiss.SwipeDismissListViewTouchListener

class DismissArrayAdapterItemListViewTouchListener(listView: ListView, arrayAdapter: ArrayAdapter<*>) : SwipeDismissListViewTouchListener(
    listView, dismissCallbacksForArrayAdapter(arrayAdapter)
) {

    companion object {
        private fun <T> dismissCallbacksForArrayAdapter(arrayAdapter: ArrayAdapter<T>): DismissCallbacks {
            return object : DismissCallbacks {
                override fun canDismiss(position: Int): Boolean {
                    return position < arrayAdapter.count
                }

                override fun onDismiss(listView: ListView?, reverseSortedPositions: IntArray) {
                    for (position in reverseSortedPositions) {
                        arrayAdapter.remove(arrayAdapter.getItem(position))
                    }
                    arrayAdapter.notifyDataSetChanged()
                }
            }
        }
    }
}